connectors/rdbms/r2dbc/src/test/kotlin/io/github/gabfssilva/river/r2dbc/R2dbcFlowExtTest.kt

```diff
@@ -0,0 +1,67 @@
+package io.github.gabfssilva.river.r2dbc
+
+import io.kotest.core.spec.style.FeatureSpec
+import io.kotest.data.forAll
+import io.kotest.data.row
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.types.shouldBeInstanceOf
+import io.r2dbc.spi.ConnectionFactories
+import io.r2dbc.spi.Result
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.asFlow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.count
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.reactive.awaitFirst
+
+class R2dbcFlowExtTest : FeatureSpec({
+    feature("Test statement execution") {
+        val connection = h2Client.awaitFirst()
+
+        scenario("Test all type of statements") {
+
+            forAll(
+                row("INSERT INTO test (id) VALUES ('%s')"),
+                row("SELECT * FROM test WHERE id = '%s'"),
+                row("UPDATE test set id = id WHERE id = '%s'"),
+                row("DELETE FROM test WHERE id = '%s'")
+            ) { statement: String ->
+
+                val flow = (1..10)
+                    .asFlow()
+                    .map {
+                        connection.createStatement(statement.format(it))
+                    }
+
+                val result = connection.executeStatementFlow(flow)
+
+                result.count() shouldBe 10
+                result.shouldBeInstanceOf<Flow<Result>>()
+            }
+        }
+
+        scenario("Batch flow"){
+            val size: Int = 10
+            val flow = (1..size).asFlow()
+            val inserts = flow.map { "INSERT INTO test (id) VALUES ('%s')".format(it) }
+            val selects = flow.map { "SELECT * FROM test WHERE id = '%s'".format(it) }
+
+
+            connection.executeBatchedStatementFlow(
+                inserts,
+                connection.createBatch()
```

Comment:
@gabfssilva I didn't like that I need to create new createBatch all of times that I want to use this fun. I would like to know if you have other approach to create this batch internally 😬